File: src/main/java/katrix/journeyToGensokyo/client/render/RenderHellRaven.java
/**
 * This class was created by <Katrix>. It's distributed as
 * part of the Journey To Gensokyo Mod. Get the Source Code in github:
 * https://github.com/Katrix-/JTG
 *
 * Journey To Gensokyo is Open Source and distributed under the
 * a modifed Botania license: https://github.com/Katrix-/JTG/blob/master/LICENSE.md
 */

package katrix.journeyToGensokyo.client.render;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import katrix.journeyToGensokyo.client.model.ModelHellRaven;
import katrix.journeyToGensokyo.lib.LibMod;
import net.minecraft.client.renderer.entity.RenderLiving;
import net.minecraft.entity.Entity;
import net.minecraft.util.ResourceLocation;

@SideOnly(Side.CLIENT)
public class RenderHellRaven extends RenderLiving {

	ResourceLocation texture = new ResourceLocation(LibMod.MODID.toLowerCase(), "textures/entity/mob/hellRaven.png");

	public RenderHellRaven() {
		super(new ModelHellRaven(), 0.25F);
	}

	@Override
	protected ResourceLocation getEntityTexture(Entity entity) {
		return texture;
	}
}

File: src/main/scala/net/katsstuff/journeytogensokyo/client/render/RenderHellRaven.scala
/*
 * This class was created by <Katrix>. It's distributed as
 * part of the Journey To Gensokyo Mod. Get the Source Code in github:
 * https://github.com/Katrix-/JTG
 *
 * Journey To Gensokyo is Open Source and distributed under the
 * a modifed Botania license: https://github.com/Katrix-/JTG/blob/devDanmakuCore/LICENSE.md
 */
package net.katsstuff.journeytogensokyo.client.render

import net.katsstuff.journeytogensokyo.client.model.ModelHellRaven
import net.katsstuff.journeytogensokyo.entity.living.EntityHellRaven
import net.katsstuff.journeytogensokyo.lib.LibMod
import net.minecraft.client.renderer.entity.{RenderLiving, RenderManager}
import net.minecraft.util.ResourceLocation

class RenderHellRaven(renderManager: RenderManager)
    extends RenderLiving[EntityHellRaven](renderManager, ModelHellRaven, 0.5F) {
  protected def getEntityTexture(entity: EntityHellRaven) =
    new ResourceLocation(LibMod.Id, "textures/entity/mob/hell_raven.png")
}
